Update developer documentation. (#888)
knative-sandbox/kn-plugin-func: docs/DEVELOPMENT.md

When a new Function is created, a few files are placed in the new Function's directory.  This includes source code illustrating a minimal Function of the requested type (language runtime and function signature) as well as Function metadata in a `func.yaml` file.

The source of these templates is `./templates`; a directory subdivided by language and template name.
For example, the Go HTTP template is located in `./templates/go/http`.
The client library and CLI are self-contained
by encoding this directory as a ZIP byte array in the `zz_filesystem_generated.go` file.
Therefore, any updates to templates requires re-generating this file.

> When changes are made to a template's source code,

Reviewer: Nit, since you're editing the docs already - this is a bit more accurate.

```suggestion
When changes are made to files in the `./templates` directory,
```

regenerate `zz_filesystem_generated.go` by running `make zz_filesystem_generated.go`.
It is also important to run the unit tests of the template modified.
For example, to run the unit tests of the Go templates, use `make test-go`.
For a list of available make targets, use `make help`.

**Tip**:

> Put following script to your `.git/hooks/pre-commit`.

Reviewer: Nit
```suggestion
Put the following script in your `.git/hooks/pre-commit` file.
```
